doc/user_doc.md

```diff
@@ -0,0 +1,415 @@
+KVDK
+=======
+
+KVDK(Key-Value Development Kit) is a Key-Value store for Persistent memory(PMEM). 
+
+KVDK supports both sorted and unsorted KV-Pairs.
+
+Code snippets in this user documents are from `./examples/tutorial.cpp`, which is built as `./build/dbtutorials`.
+
+## Open a KVDK instance
+
+A KVDK instance corresponds to a PMEM directory mounted under linux system. Keys and values
+are stored in PMEM under this path and are indexed by HashTable which is stored in DRAM.
+HashTable is not persisted on PMEM and is reconstructed with information on PMEM.
+
+The following code shows how to create a KVDK instance if none exist
+or how to reopen an existing KVDK instance at the path supplied.
+(In the following example, PMEM is mounted as /mnt/pmem0/ and KVDK is named tutorial_kvdk_example.)
+
+```c++
+#include "kvdk/engine.hpp"
+#include "kvdk/namespace.hpp"
+#include <string>
+#include <thread>
+#include <vector>
+#include <cassert>
+#include <random>
+#include <algorithm>
+
+#define DEBUG   // For assert
+
+int main()
+{
+  kvdk::Status status;
+
+  // Initialize a KVDK instance.
+  kvdk::Engine *db = nullptr;
+  kvdk::Configs db_configs;
+  {
+    // Configure for a tiny KVDK instance. Approximately 10MB /mnt/pmem0/ space
+    // is needed. Please refer to "Configuration" section in user documentation
+    // for details.
+    db_configs.pmem_file_size = (1ull << 20);
+    db_configs.pmem_segment_blocks = (1ull << 8);
+    db_configs.num_hash_buckets = (1ull << 10);
+  }
+  // The KVDK instance is mounted as DB under /mnt/pmem0/
+  // Modify this path if necessary.
+  std::string db_path{"/mnt/pmem0/tutorial_kvdk_example"};
```

Comment:
Since we don't call it db now, maybe we should change db to instance or engine, e.g. db_path -> instance_path, db_configs->configs